datalake/src/main/java/com/sequenceiq/datalake/service/sdx/CloudbreakFlowService.java

```diff
@@ -42,6 +43,11 @@ public FlowLogResponse getLastCloudbreakFlowChainId(SdxCluster sdxCluster) {
                 flowEndpoint.getFlowLogsByFlowId(flowId));
     }
 
+    public boolean getFlowResultByFlowId(SdxCluster sdxCluster) {
```

Comment:
Generally what we need to do is to check the flow state. Did it complete, did it fail. SdxCluster has

```java
    private String lastCbFlowChainId;

    private String lastCbFlowId;
```

columns. Based on these we should be able to use 

- `com.sequenceiq.flow.api.FlowEndpoint#hasFlowRunningByChainId`
- `com.sequenceiq.flow.api.FlowEndpoint#hasFlowRunningByFlowId`

endpoints and figure out the state (completed / failed). We should not poll the entire flow log and make decisions based on it. FlowEndpoint already provides a solution to do it better.

Additionally the below logic wont work if the flow first failed, then retried (there will be FAILED states).

```java
return flowLogsByFlowId.stream().allMatch(fl -> StateStatus.SUCCESSFUL.equals(fl.getStateStatus()));
```

Reply:
Yes, you are right, will change it.

I used it because of a special use-case: it is possible that a flow state fails but the whole flow is not failed. But that is a special thing, should not be generalized.

Reply:
Yes after a retry there will be failed flow states but it can be successful flow so it's better to handle it. I afraid that developers wont go into the details just assume it's working but actually it's not. Other thing is that it would be better to give it a name like isLastFlowSuccessful(cluster) or something.